metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/utilities/PerformanceMetricsUtilities.java

```diff
@@ -0,0 +1,234 @@
+package eu.europeana.metis.performance.metric.utilities;
+
+import eu.europeana.metis.core.dao.PluginWithExecutionId;
+import eu.europeana.metis.core.dao.WorkflowExecutionDao;
+import eu.europeana.metis.core.dataset.Dataset;
+import eu.europeana.metis.core.workflow.WorkflowExecution;
+
+import eu.europeana.metis.core.workflow.plugins.AbstractExecutablePlugin;
+import eu.europeana.metis.core.workflow.plugins.AbstractMetisPlugin;
+import eu.europeana.metis.core.workflow.plugins.ExecutablePlugin;
+import eu.europeana.metis.core.workflow.plugins.PluginType;
+import eu.europeana.metis.performance.metric.dao.MongoMetisCoreDao;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.TreeSet;
+import java.util.Optional;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class PerformanceMetricsUtilities {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CSVUtilities.class);
+    private static final SimpleDateFormat SIMPLE_DATE_TIME_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private static final SimpleDateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+
+    private final MongoMetisCoreDao mongoMetisCoreDao;
+
+    public PerformanceMetricsUtilities(MongoMetisCoreDao mongoMetisCoreDao) {
+        this.mongoMetisCoreDao = mongoMetisCoreDao;
+    }
+
+    public List<String> getDateForMetric1(LocalDateTime startLocalDateTime, LocalDateTime endLocalDateTime){
```

Comment:
My suggestion would be to separate logic of Metric1 and Metric2 (metric3, metric4, etc) in two (or more) different classes, so in the future giving maintenance will be easy for us. I see this class can become quite big if we leave all code here.

Reply:
That's a good point, I'll follow  this suggestion